src/programs/util/ectrans_field_api_helper.F90

```diff
@@ -0,0 +1,480 @@
+  class (field_3rb), pointer :: u_ew, v_ew    ! grid-point u and derivatives
+
+  class (field_4rb), pointer :: scalar3      ! grid-point scalar fields
+  class (field_4rb), pointer :: scalar3_ew   ! grid-point scalar fields derivatives ew
+  class (field_4rb), pointer :: scalar3_ns   ! grid-point scalar fields derivatives ns
+
+  class (field_3rb), pointer :: scalar2      ! grid-point surfacic scalar fields
+  class (field_3rb), pointer :: scalar2_ew   ! grid-point surfacic scalar fields derivatives ew
+  class (field_3rb), pointer :: scalar2_ns   ! grid-point surfacic scalar fields derivatives ns
+
+  class (field_3rb), pointer :: scalar   ! grid-point scalar fields as a single field
+  class (field_3rb), pointer :: scalar_ew! grid-point scalar fields derivatives ew
+  class (field_3rb), pointer :: scalar_ns! grid-point scalar fields derivatives ns
+end type wrapped_fields
+
+type fields_lists
+
+! List of field lists that will be used as parameter to inv_trans_field_api and dir_trans_field_api
+
+  type (field_basic_ptr), allocatable :: u (:), v (:)                ! grid-point u and v fields
+  type (field_basic_ptr), allocatable :: scalar (:)                  ! grid-point scalar fields
+  type (field_basic_ptr), allocatable :: spvor (:), spdiv (:)        ! spectral vorticity and divergence
+  type (field_basic_ptr), allocatable :: vor (:), div (:)            ! grid-point vorticity and diverence
+  type (field_basic_ptr), allocatable :: spscalar (:)                ! spectral scalar fields
+  type (field_basic_ptr), allocatable :: u_ew (:), v_ew (:)            ! grid-point u and derivatives we
+  type (field_basic_ptr), allocatable :: scalar_ns (:), scalar_ew (:)  ! grid space scalar derivatives ns and ew
+  end type fields_lists
+
+contains
+
+subroutine wrap_benchmark_fields_zgp(ywflds, lvordiv, lscders, luvders,&
+                                   & zspvor, zspdiv, zspscalar, zgp)
+
+  ! Wrap the arrays given as input in field API objects
+
+    type(wrapped_fields), intent(inout) :: ywflds
+    logical, intent(in) :: lvordiv
+    logical, intent(in) :: lscders
+    logical, intent(in) :: luvders
+    real(kind=jprb), intent(in) :: zspvor(:,:)
+    real(kind=jprb), intent(in) :: zspdiv(:,:)
+    real(kind=jprb), intent(in) :: zspscalar(:,:)
+    real(kind=jprb), intent(in) :: zgp (:,:,:)
+
+    integer :: inum_wind_fields,  inum_sc_2d_fields
+    integer :: ioffset
+
+    inum_wind_fields = size(zspvor,1)
+    inum_sc_2d_fields = size(zspscalar,1)
+    ! spectral vector fields
+    call field_new(ywflds%spvor,  data=zspvor(:,:))
+    call field_new(ywflds%spdiv,  data=zspdiv(:,:))
+
+    ! spectral scalar fields
+    if (inum_sc_2d_fields > 0) then
+      call field_new(ywflds%spscalar, data=zspscalar(:,:))
+    endif
+    ioffset = 1
+
+    ! grid-point vector fields
+
+    ! gridpoint vector field vorticitity and divergence
+    if (lvordiv) then
+        call field_new(ywflds%vor, data=zgp(:, ioffset:ioffset+inum_wind_fields-1, :))
```

Comment:
I think this might be an error. You're using `inum_wind_fields` to set the "fields" dimension of `zgp`, i.e. the total number of grid point fields. But `inum_wind_fields` is computed from the first dimension of `zspvor`. In spectral space the fields dimension is partitioned over the V set, so these two aren't the same.